contracts/contracts/eosio.system/src/delegate_bandwidth.cpp

```diff
@@ -295,6 +297,16 @@ namespace eosiosystem {
       check( _gstate.total_activated_stake >= min_activated_stake,
              "cannot undelegate bandwidth until the chain is activated (at least 15% of all tokens participate in voting)" );
 
+      if(is_block_producer(from))
+      {
+          del_bandwidth_table     del_tbl( _self, from.value );
+          auto itr = del_tbl.find( from.value );
```

Comment:
You can do this with one line. Smth like this:
const auto& del = del_tbl.get( from.value, "user has no resources" );